src/jit_channel/channel_manager.rs

```diff
@@ -0,0 +1,1239 @@
+					code: LSPS2_BUY_REQUEST_INVALID_VERSION_ERROR_CODE,
+					message: format!("version {} is not supported", params.version),
+					data: Some(format!("Supported versions are {:?}", SUPPORTED_SPEC_VERSIONS)),
+				}),
+			);
+			return Err(LightningError {
+				err: format!("client requested unsupported version {}", params.version),
+				action: ErrorAction::IgnoreAndLog(Level::Info),
+			});
+		}
+
+		if let Some(payment_size_msat) = params.payment_size_msat {
+			if payment_size_msat < self.min_payment_size_msat {
+				self.enqueue_response(
+					*counterparty_node_id,
+					request_id,
+					LSPS2Response::BuyError(ResponseError {
+						code: LSPS2_BUY_REQUEST_PAYMENT_SIZE_TOO_SMALL_ERROR_CODE,
+						message: "payment size is below our minimum supported payment size"
+							.to_string(),
+						data: None,
+					}),
+				);
+				return Err(LightningError {
+					err: "payment size is below our minimum supported payment size".to_string(),
+					action: ErrorAction::IgnoreAndLog(Level::Info),
+				});
+			}
+
+			if payment_size_msat > self.max_payment_size_msat {
+				self.enqueue_response(
+					*counterparty_node_id,
+					request_id,
+					LSPS2Response::BuyError(ResponseError {
+						code: LSPS2_BUY_REQUEST_PAYMENT_SIZE_TOO_LARGE_ERROR_CODE,
+						message: "payment size is above our maximum supported payment size"
+							.to_string(),
+						data: None,
+					}),
+				);
+				return Err(LightningError {
+					err: "payment size is above our maximum supported payment size".to_string(),
+					action: ErrorAction::IgnoreAndLog(Level::Info),
+				});
+			}
+
+			match compute_opening_fee(
+				payment_size_msat,
+				params.opening_fee_params.min_fee_msat,
+				params.opening_fee_params.proportional,
+			) {
+				Some(opening_fee) => {
+					if opening_fee >= payment_size_msat {
+						self.enqueue_response(
+							*counterparty_node_id,
+							request_id,
+							LSPS2Response::BuyError(ResponseError {
+								code: LSPS2_BUY_REQUEST_PAYMENT_SIZE_TOO_SMALL_ERROR_CODE,
+								message: "payment size is too small to cover the opening fee"
+									.to_string(),
+								data: None,
+							}),
+						);
+						return Err(LightningError {
+							err: "payment size is too small to cover the opening fee".to_string(),
+							action: ErrorAction::IgnoreAndLog(Level::Info),
+						});
+					}
+				}
+				None => {
+					self.enqueue_response(
+						*counterparty_node_id,
+						request_id,
+						LSPS2Response::BuyError(ResponseError {
+							code: LSPS2_BUY_REQUEST_PAYMENT_SIZE_TOO_LARGE_ERROR_CODE,
+							message: "overflow error when calculating opening_fee".to_string(),
+							data: None,
+						}),
+					);
+					return Err(LightningError {
+						err: "overflow error when calculating opening_fee".to_string(),
+						action: ErrorAction::IgnoreAndLog(Level::Info),
+					});
+				}
+			}
+		}
+
+		// TODO: if payment_size_msat is specified, make sure our node has sufficient incoming liquidity from public network to receive it.
```

Comment:
This is mentioned in the spec but doesn't really feel like it belongs in the library.  I think we probably need to expose a failure flow for the LSP to call into as an alternative to `invoice_parameters_generated` when they receive a BuyRequest event.

This way if they decide they can't honor/fulfill the request for whatever reason they have a way to bail.  We would add this spec comment to the description of BuyRequest Event for a potential reason they might fail it.